config/configload/validate_test.go

```diff
@@ -556,3 +556,89 @@ func TestPermissionMixed(t *testing.T) {
 		})
 	}
 }
+
+func TestPathAttr(t *testing.T) {
+	tests := []struct {
+		name  string
+		hcl   string
+		error string
+	}{
+		{
+			"path in endpoint: error",
+			`server {
+  endpoint "/**" {
+    path = "/a/**"
+  }
+}`,
+			"couper.hcl:3,5-9: Unsupported argument; An argument named \"path\" is not expected here. Use the \"path\" attribute in a backend block instead.",
+		},
+		{
+			"path in proxy: error",
+			`server {
+  endpoint "/**" {
+    proxy {
+      path = "/a/**"
+    }
+  }
+}`,
+			"couper.hcl:4,7-11: Unsupported argument; An argument named \"path\" is not expected here. Use the \"path\" attribute in a backend block instead.",
+		},
+		{
+			"path in referenced backend: ok",
+			`server {
+  endpoint "/**" {
+    proxy {
+      backend = "a"
+    }
+  }
+}
+definitions {
+  backend "a" {
+    path = "/a/**"
+  }
+}`,
+			"",
+		},
+		{
+			"path in refined backend: ok",
+			`server {
+  endpoint "/**" {
+    proxy {
+      backend "a" {
+        path = "/a/**"
+      }
+    }
+  }
+}
+definitions {
+  backend "a" {
+  }
+}`,
+			"",
+		},
+	}
+
+	logger, _ := logrustest.NewNullLogger()
+	log := logger.WithContext(context.TODO())
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(subT *testing.T) {
+			conf, err := LoadBytes([]byte(tt.hcl), "couper.hcl")
+			if conf != nil {
+				tmpStoreCh := make(chan struct{})
+				defer close(tmpStoreCh)
+
+				_, err = runtime.NewServerConfiguration(conf, log, cache.New(log, tmpStoreCh))
+			}
```

Comment:
What is this for?

Reply:
@johakoch @afflerbach this code further validates the given configuration (business logic). Loading the bytes just applies merging and schema validation. Handling the error should be refined instead of removing this runtime call. Things like `endpoint must define at least one x block` are not handled by schema validation.

Reply:
The new test at least is only about schema validation. So the runtime call can in fact be removed for this one.

Reply:
Also, TestAttributeObjectKeys and TestPermissionMixed test code from config load.